### PR TITLE
use v{major} as jdl sample repository

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
@DanielFran if you agree, change the default branch to v8: https://github.com/jhipster/jdl-samples/tree/v8

File: cli/download.mts [hunk 1]
```diff
@@ -63,7 +63,10 @@ export const downloadJdlFile = async (jdlFile: string, { skipSampleRepository }:
     return await downloadFile(url, jdlFile);
   } catch (error) {
     logger.info((error as any).message);
-    url = new URL(jdlFile, 'https://raw.githubusercontent.com/jhipster/jdl-samples/main/').toString();
+    // TODO replace when any v8 is released.
+    // const majorVersion = packageJson.version.split('.', 2)[0];
+    const majorVersion = '8';
```
Comment (DanielFran) [inline cli/download.mts:68]:
```suggestion
    const branchName = 'v8';
```

File: cli/download.mts [hunk 2]
```diff
@@ -63,7 +63,10 @@ export const downloadJdlFile = async (jdlFile: string, { skipSampleRepository }:
...
+    url = new URL(jdlFile, `https://raw.githubusercontent.com/jhipster/jdl-samples/v${majorVersion}/`).toString();
```
Comment (DanielFran) [inline cli/download.mts:69]:
```suggestion
    url = new URL(jdlFile, `https://raw.githubusercontent.com/jhipster/jdl-samples/${branchName}/`).toString();
```

File: cli/download.mts [hunk 3]
```diff
@@ -63,7 +63,10 @@ export const downloadJdlFile = async (jdlFile: string, { skipSampleRepository }:
     return await downloadFile(url, jdlFile);
   } catch (error) {
     logger.info((error as any).message);
-    url = new URL(jdlFile, 'https://raw.githubusercontent.com/jhipster/jdl-samples/main/').toString();
+    // TODO replace when any v8 is released.
+    // const majorVersion = packageJson.version.split('.', 2)[0];
```
Comment (DanielFran) [inline cli/download.mts:67]:
```suggestion
    // branchName = 'v' + packageJson.version.split('.', 2)[0];
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Review by DanielFran — APPROVED** [review]

**Pull request merged**